src/main/java/com/laytonsmith/core/AliasCore.java

```diff
@@ -215,127 +199,123 @@ public final void reload(MCPlayer player, String[] settings, boolean firstLoad)
+			} catch (IOException ex) {
+				Static.getLogger().log(Level.SEVERE, "Failed to reload preferences.", ex);
+				return;
+			}
+		} else {
 			if(Prefs.AllowDynamicShell()) {
 				MSLog.GetLogger().Log(MSLog.Tags.GENERAL, LogLevel.WARNING, "allow-dynamic-shell is set to true in "
-						+ CommandHelperFileLocations.getDefault().getProfilerConfigFile().getName()
+						+ fileLocations.getProfilerConfigFile().getName()
 						+ " you should set this to false, except during development.", Target.UNKNOWN);
 			}
+		}
 
-			if(parent.profiler == null || options.reloadProfiler()) {
-				parent.profiler = new Profiler(MethodScriptFileLocations.getDefault().getProfilerConfigFile());
-			}
+		MSLog.initialize(fileLocations.getConfigDirectory());
 
-			ProfilePoint extensionPreReload = parent.profiler.start("Extension PreReloadAliases call", LogLevel.VERBOSE);
+		Profiler profiler;
+		if(options.reloadProfiler()) {
 			try {
-				// Allow new-style extensions know we are about to reload aliases.
-				ExtensionManager.PreReloadAliases(options);
-			} finally {
-				extensionPreReload.stop();
+				profiler = new Profiler(fileLocations.getProfilerConfigFile());
+			} catch (IOException ex) {
+				Static.getLogger().log(Level.SEVERE, "Failed to create Profiler from config file.", ex);
+				return;
 			}
+		} else {
+			// If not reloading profiler, keep the old one.
+			profiler = staticRuntimeEnv.GetProfiler();
+		}
+
+		// Allow extensions know we are about to reload. Runs extension's onPreReloadAliases().
+		ProfilePoint extensionPreReload = profiler.start("Extension PreReloadAliases call", LogLevel.VERBOSE);
+		try {
+			ExtensionManager.PreReloadAliases(options);
+		} finally {
+			extensionPreReload.stop();
+		}
 
-			ProfilePoint shutdownHooks = parent.profiler.start("Shutdown hooks call", LogLevel.VERBOSE);
+		Profiles profiles;
+		try {
+			profiles = new ProfilesImpl(fileLocations.getProfilesFile());
+		} catch (IOException | Profiles.InvalidProfileException ex) {
+			MSLog.GetLogger().e(MSLog.Tags.GENERAL, ex.getMessage(), Target.UNKNOWN);
+			return;
+		}
+
+		PersistenceNetwork persistenceNetwork;
+		if(options.reloadPersistenceConfig()) {
+			ProfilePoint persistenceConfigReload = profiler.start("Reloading persistence configuration", LogLevel.VERBOSE);
 			try {
-				StaticLayer.GetConvertor().runShutdownHooks();
+				DataSourceFactory.DisconnectAll();
+				ConnectionMixinFactory.ConnectionMixinOptions opts = new ConnectionMixinFactory.ConnectionMixinOptions();
+				opts.setWorkingDirectory(fileLocations.getConfigDirectory());
+				persistenceNetwork = new PersistenceNetworkImpl(fileLocations.getPersistenceConfig(),
+						new URI("sqlite://" + fileLocations.getDefaultPersistenceDBFile().getCanonicalFile().toURI()
+								.getRawSchemeSpecificPart().replace('\\', '/')), opts);
+			} catch (DataSourceException | URISyntaxException | IOException ex) {
+				Static.getLogger().log(Level.SEVERE, null, ex);
+				return;
```

Comment:
Same here. Maybe we ought to have an option to "stop server startup on failure", since one might rely on CH for some security, and would rather the server not start up if there's a misconfiguration, rather than startup without CH.

Reply:
Ya, it's a tough one. We have a halt-on-failure preferences option. We could make it safely shutdown the server if it's on firstload, keep the old profiles/etc if not on reload.